docs/crocks/Pred.md

```diff
@@ -0,0 +1,57 @@
+# Pred
+
+`Pred a`
+
+--
+
+--
+
+```js
+--
+```
+
+`Pred` exposes the following functions on the constructor and instance:
+
+| Constructor | Instance |
+|:---|:---|
+| [`empty`](#empty), [`type`](#type) | [`concat`](#concat), [`contramap`](#contramap), [`empty`](#empty),  [`inspect`](#inspect), [`runWith`](#runWith), [`type`](#type), [`value`](#value) |
+
+## Constructor
+
+### empty
+
+`Pred m => () ~> m ()`
+
+### type
+
+`() -> String`
+
+## Instance
+
+### concat
+
+`Pred m => m a ~> a -> m a`
+
+### contramap
+
+`Pred m => m a ~> a -> m ()`
+
+### empty
+
+`Pred m => () ~> m ()`
+
+### inspect
+
+`() => String`
+
+### runWith
+
+`Pred m => m a ~> () -> a`
+
+### type
+
+`() -> String`
+
+### value
+
+`Pred m => m b ~> () -> b`
```

Comment:
So this is a strange one, and it is only there to work with Monoidial reduction.
This is going to extract the inner function. So really this returns the function `(a -> Boolean)`.
So this should look like:

`Pred m => m a ~> () -> (a -> Boolean)`

we do not auto-curry `value` that is why it is not `() -> a -> Boolean`